### ISSUE TITLE
Task execution throws exception

### ISSUE THREAD

**Issue by MortenChristiansen** [brthor/Gofer.NET#25]
Since version 1.0.0-beta-00110, our task execution code has stopped working. You can see how we do it, as well as the resulting exception, below.

```
await _taskClient.TaskScheduler.AddRecurringTask(() => Execute(command, _logger, _bus), TimeSpan.FromSeconds(schedule.Interval), taskName);

private static void Execute(Type command, IBumaLogger logger, ICommandSenderBus bus)
{
    logger.Debug($"Executing scheduled command {command.FullName}");
    try
    {
        var commandInstance = (IScheduledCommand)Activator.CreateInstance(command);
        bus.SendAsync(commandInstance);
    }
    catch (Exception e)
    {
        logger.Error($"Error executing scheduled command {command.FullName}", e);
    }
}

Task Error: Buma.Host, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null.Execute threw an exception. (Task Id: 445f31a4-e1c7-4f26-8cf1-433577506137)
Sample.Host> System.ArgumentException: Object of type 'System.String' cannot be converted to type 'System.Type'.
Sample.Host>    at System.RuntimeType.TryChangeType(Object value, Binder binder, CultureInfo culture, Boolean needsSpecialCast)
Sample.Host>    at System.Reflection.MethodBase.CheckArguments(Object[] parameters, Binder binder, BindingFlags invokeAttr, CultureInfo culture, Signature sig)
Sample.Host>    at System.Reflection.RuntimeMethodInfo.InvokeArgumentsCheck(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
Sample.Host>    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
Sample.Host>    at Gofer.NET.Utils.TaskInfo.ExecuteTask()
Sample.Host>    at Gofer.NET.TaskClient.Listen()
Sample.Host> Debug: Forwarded 1 lines
```

**Comment by brthor** [comment]
Thanks for filing this bug @MortenChristiansen , I am beginning to investigate.

**Comment by brthor** [comment]
I managed to reproduce this in a simple test, and am working on finding a fix.

**Comment by brthor** [comment]
This issue is fixed in `1.0.0-beta-00127`, by PR #26 

We have test coverage for this scenario now as well.

Thanks again @MortenChristiansen for filing this, if you have any more issues post-update feel free to comment and I will reopen it.
